groth16/examples/sha256.rs

```diff
@@ -25,58 +24,69 @@ use groth16::proving_key::PackedProvingKeyShare;
 #[cfg(feature = "parallel")]
 use rayon::prelude::*;
 
-async fn dsha256<E: Pairing, Net: MpcNet>(
+async fn dsha256<E, Net>(
     pp: &PackedSharingParams<E::ScalarField>,
     crs_share: &PackedProvingKeyShare<E>,
+    qap_share: qap::PackedQAPShare<
+        E::ScalarField,
+        Radix2EvaluationDomain<E::ScalarField>,
+    >,
     a_share: &[E::ScalarField],
     ax_share: &[E::ScalarField],
-    h_share: &[E::ScalarField],
     net: &Net,
-) -> (E::G1, E::G2, E::G1) {
-    debug!(
-        "s:{}, v:{}, h:{}, w:{}, u:{}, a_share:{}, h_share:{}",
-        crs_share.s.len(),
-        crs_share.v.len(),
-        crs_share.h.len(),
-        crs_share.w.len(),
-        crs_share.u.len(),
-        a_share.len(),
-        h_share.len()
-    );
-
+) -> (E::G1, E::G2, E::G1)
+where
+    E: Pairing,
+    Net: MpcNet,
+{
+    let h_share = ext_wit::h(qap_share, pp, &net).await.unwrap();
     let msm_section = start_timer!(|| "MSM operations");
     // Compute msm while dropping the base vectors as they are not used again
     let compute_a = start_timer!(|| "Compute A");
-    let pi_a_share: E::G1 =
-        dmsm::d_msm(&crs_share.s, a_share, pp, net, MultiplexedStreamID::One)
-            .await
-            .unwrap();
+    let pi_a_share = groth16::prove::A::<E> {
+        L: Default::default(),
+        N: Default::default(),
+        r: E::ScalarField::zero(),
```

Comment:
Should we at least change these to `E::ScalarField::one()`, making it zero, will nullify the value in question in the exponent. Of course we should just handle this randomness properly.

Reply:
We are aiming to add the randomness eventually once we start it in the gadget and the full design is complete.